packages/webapp/src/components/Finances/Transaction/ExpandedContent/LabourTable.jsx

```diff
@@ -0,0 +1,173 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import styles from './styles.module.scss';
+import { useTranslation } from 'react-i18next';
+import EnhancedTable from '../../../Table/v2';
+import StateTab from '../../../RouterTab/StateTab';
+import { useCurrencySymbol } from '../../../../containers/hooks/useCurrencySymbol';
+
+export default function LabourTable({ transaction }) {
+  const { t } = useTranslation();
+  const currencySymbol = useCurrencySymbol();
+
+  const [activeTab, setActiveTab] = useState('employee');
+
+  const employeeTotalHours = useMemo(
+    () => calculateTotalHours(transaction.items.EMPLOYEE),
+    [transaction],
+  );
+
+  const taskTotalHours = useMemo(
+    () => calculateTotalHours(transaction.items.TASK_TYPE),
+    [transaction],
+  );
```

Comment:
@antsgar 
I am curious about the approach you would take.
I would create variables for the activeTab only, so I would combine `employeeTotalHours` and `taskTotalHours` as `totalHours`, and I would create `columns` to return columns for the activeTab.
It would be less code and calculations for inactive tab would not happen.

However, Joyce's approach is good because re-calculation will not happen by changing the tab.

How would you make a decision?

Reply:
Both are okay IMO! Although even though there are things in common between the two sets of columns, there's also enough differences that I think bundling them together might cause for the code to have a bunch of conditions which I think would make it a bit harder to read that it is right now. I wouldn't worry too much about the calculations in this case because the arrays are super straight forward and they should take virtually no time to be calculated